Automattic/wp-calypso
# Add plans grid to Woo Express plans

## Proposed Changes

This PR ensures that we display the Woo Express plans grid to users who have purchased a Woo Express Essential or Woo Express Performance plan, though the new display is only available when the `plans/wooexpress-small` feature flag is enabled _OR_ the site already has a Woo Express Essential plan subscription.

Furthermore, the PR includes the following changes:
* Rename the existing `WooExpressMediumPage` to `WooExpressPlansPage`, and refactor it to support both small and medium plans
* Update the styles for the `WooExpressPlans` component to remove the branded fonts in the plans grid for this context as well as the free trial context
* Ensure that we correctly allow the Performance plan to be considered an upgrade for the Essential plan
* Ensure we have the Essential monthly plan in our monthly plans list
* Refine the custom labels added in #75229 to use them only when the user doesn't have the plan already (especially a different billing interval)
* Use the same rounding as the plans grid for discounts

For now the PR does not tackle the following items:
* Updating the feature list
* Adding the new plans grid to the expired trial page

### Screenshots

#### Performance plan
<img width="1195" alt="Screenshot 2023-04-03 at 22 24 28" src="https://user-images.githubusercontent.com/3376401/229620514-162e7932-e2f7-460a-8e25-8fd5def9c712.png">
<img width="1195" alt="Screenshot 2023-04-03 at 22 25 23" src="https://user-images.githubusercontent.com/3376401/229620532-70e4c524-ca57-4180-a46b-0ea189fcf0c8.png">

#### Essential plan
<img width="1195" alt="Screenshot 2023-04-03 at 22 25 18" src="https://user-images.githubusercontent.com/3376401/229620542-cb4607ca-e323-4a2d-9ffd-b2070bdbaded.png">
<img width="1195" alt="Screenshot 2023-04-03 at 22 25 02" src="https://user-images.githubusercontent.com/3376401/229620547-ea6991b5-578e-41a2-8cfa-f1309300c1e8.png">

## Testing Instructions

* Ensure that you have an eCommerce trial site
* For that site, navigate to _Upgrades_ -> _Plans_ on a local dev build or the Calypso.live build (`/plans/:siteSlug`)
* Ensure that the `plans/wooexpress-small` feature flag is enabled by adding `?flags=plans/wooexpress-small` to the URL
* Verify that the fonts for the plan names and prices in the grid are sans-serif fonts
* Work through the purchase flow for the monthly Essential plan
* After completing the purchase, navigate back to _Upgrades_ -> _Plans_
* Verify that you see the pricing grid with both the Essential and Performance plans, and can switch between monthly and annual plans
* Verify that the discount in the header matches the discount in the pricing grid
* Verify that you are shown the option to upgrade to the annual Essential plan and both the monthly and annual Performance plans
* Ensure that the `plans/wooexpress-small` feature flag is **disable** by adding `?flags=-plans/wooexpress-small` to the URL
* Verify that you still see the pricing grid
* Work through the purchase flow for the monthly Performance plan
* After completing the purchase, navigate back to _Upgrades_ -> _Plans_
* Ensure that the `plans/wooexpress-small` feature flag is enabled by adding `?flags=plans/wooexpress-small` to the URL
* Verify that you see the pricing grid displayed
* Verify that the Essential plan CTAs show _Downgrade_ but require support contact
* Verify that you can switch between annual and monthly terms
* Verify that the annual Performance plan is shown as an upgrade option

## Pre-merge Checklist



- [N/A] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
